lib/frame.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require_relative './roll'
+
+class Frame
+  attr_reader :pins_left, :result, :score
+
+  def initialize
+    @rolls = []
+    @pins_left = 10
+    @result = ''
+    @score = 0
+  end
+
+  def roll(pins, bonus = false)
```

Comment:
Nicely written method. In only a few lines you've included a lot of code functionality. Taking a bonus parameter preset to false, ternary conditional, factored in an error message case, setting a current roll instance using a new instance of the Roll class.